client/chat/route.tsx

```diff
@@ -1,30 +1,56 @@
-import React from 'react'
-import { useRoute } from 'wouter'
-import { makeSbChannelId } from '../../common/chat'
+import React, { Suspense } from 'react'
+import { Route, RouteProps, Switch } from 'wouter'
+import { makeSbChannelId, SbChannelId } from '../../common/chat'
+import { hasAnyPermission } from '../admin/admin-permissions'
 import { replace } from '../navigation/routing'
+import { LoadingDotsArea } from '../progress/dots'
+import { useAppSelector } from '../redux-hooks'
 import { ConnectedChatChannel } from './channel'
+import { ChannelList } from './channel-list'
+import { CreateChannel } from './create-channel'
 
-export function ChannelRouteComponent(props: { params: any }) {
-  const [matches, params] = useRoute('/chat/:channelId/:channelName')
+const LoadableChatAdmin = React.lazy(async () => ({
+  default: (await import('./admin')).ChatAdmin,
+}))
+
+export function ChannelRoute({
+  component: Component,
+  ...rest
+}: Omit<RouteProps, 'component'> & {
+  component: React.ComponentType<{ channelId: SbChannelId; channelName: string }>
+}) {
+  return (
+    <Route<{ channelId: string; channelName: string }> {...rest}>
+      {params => {
+        const channelIdNum = Number(params.channelId)
+        if (isNaN(channelIdNum)) {
+          queueMicrotask(() => {
+            replace('/')
+          })
+          return null
+        }
 
-  if (!matches) {
-    queueMicrotask(() => {
-      replace('/')
-    })
-    return null
-  }
-  const channelIdNum = Number(params.channelId)
-  if (isNaN(channelIdNum)) {
-    queueMicrotask(() => {
-      replace('/')
-    })
-    return null
-  }
+        return (
+          <Component channelId={makeSbChannelId(channelIdNum)} channelName={params.channelName} />
+        )
+      }}
+    </Route>
+  )
+}
+
+export function ChannelRouteComponent(props: { params: any }) {
+  // TODO(2Pac): Add a separate permission for managing chat channels
+  const isAdmin = useAppSelector(s => hasAnyPermission(s.auth, 'moderateChatChannels'))
 
   return (
-    <ConnectedChatChannel
-      channelId={makeSbChannelId(channelIdNum)}
-      channelName={params!.channelName}
-    />
+    <Suspense fallback={<LoadingDotsArea />}>
+      <Switch>
+        {isAdmin ? <Route path='/chat/admin/:rest*' component={LoadableChatAdmin} /> : <></>}
```

Comment:
I would recommend always having a route for this and just rendering something that says they don't have permission to access it or something. Not having a route all the time in Leagues has caused problems and needs to be fixed